tests/src/AuthorizationTest.idr

```diff
@@ -0,0 +1,28 @@
+module AuthorizationTest
+
+import Network.HTTP
+import Control.Monad.Error.Either
+import Data.String.Base64
+
+%default partial
+
+export
+test_base64_encoding_decoding : EitherT String IO ()
+test_base64_encoding_decoding =
+  let user           = "aladdin"
+      passwd         = "opensesame"
+      (_, basicauth) = applyBasicAuth user passwd
+      basicauth'     = base64DecodeString basicauth
+    in case basicauth' of
+         Nothing          =>
+           idris_crash "couldn't base64 decode string."
```

Comment:
I guess this is the one that fails ? Can you add different error messages for the two tests? So its easier to see what test fails? And i dont think using idris crash is how the other tests are designed to fail. I could be wrong of course, but i dont remember seing a test crash like this before.

Reply:
@jarlah There was a small oversight on my part here (easy fix):

`basicauth` here is `Basic [base64encodedstring]`, but I was accidentally treating it as `[base64encodedstring]`.

Thus I needed to `break` `basicauth` apart, and call `base64DecodeString` only on the `[base64encodedstring]` part.

Reply:
Hmm, it failed again, let me take a look.

Reply:
In a test, code like this is fine. In normal code i would have asked to create a method to keep my brain from trying to decode it. 🤣

Reply:
Are you running tests locally btw

Reply:
@jarlah 

For some reason, I have not been able to run the tests locally (probably why I keep pushing failing code) even though I successfully build the library and the test suite, here is what I am seeing:

```
matthewmosior@matthewmosior-B550-AORUS-ELITE-AX-V2:tests$ pack run
Exception: (while loading libidristls.so) libidristls.so: cannot open shared object file: No such file or directory
[ fatal ] Error when executing system command.
          Command: "/home/matthewmosior/Software/git/idris2-http/tests/build/exec/runtests"
          Error code: 255
```

Super weird though, I definitely have `tls` installed via `pack` on my system.

Reply:
i would guess its a cache thing, because workflow does not do any hacks. Im trying to figure out how to clean pack cache or pack dependencies .. just removing build under root and test doesnt make it rebuild the build folders when i rerun tests.

Reply:
probably $HOME/.pack or something. I will look into mine and see if it has "hacks"

Reply:
no hacks in user config. You could try pack update-db? i destroyed my setup now and its bootstrapping pack .. taking ages

Reply:
if rebuilding you should get these two when running tests 

[ warning ] Package base64 uses custom build hooks. Continue (yes/*no)?
yes
[ warning ] Package tls uses custom build hooks. Continue (yes/*no)?
yes

Reply:
@Matthew-Mosior after full rebuild of pack and with default contents in .pack/user/pack.toml, i can still run the tests. Maybe remove stuff from .pack/.cache/tls?

Reply:
I will test the stuff in repl later today ;) you could also use pack repl and test the individual operations

Reply:
@jarlah 

I've fixed this function, was really silly oversight:

It had to do with the `break` function.

`break` wasn't stripping the leading white space in `Basic YWxhZGRpbjpvcGVuc2VzYW1l`, causing `base64DecodeString` to be called on completely erroneous input (should have been called on `YWxhZGRpbjpvcGVuc2VzYW1l`, not ` YWxhZGRpbjpvcGVuc2VzYW1l`).

Using `split` accomplished what needed to be done here.